src/Microsoft.ML.Transforms/Text/WordEmbeddingsTransform.cs

```diff
@@ -0,0 +1,448 @@
+        }
+
+        public override void Save(ModelSaveContext ctx)
+        {
+            Host.CheckValue(ctx, nameof(ctx));
+            ctx.CheckAtModel();
+            ctx.SetVersionInfo(GetVersionInfo());
+
+            SaveBase(ctx);
+            ctx.Writer.WriteBoolByte(_customLookup);
+            if (_customLookup)
+                ctx.SaveString(_modelFileNameWithPath);
+            else
+                ctx.Writer.Write((uint)_modelKind);
+        }
+
+        protected override ColumnType GetColumnTypeCore(int iinfo)
+        {
+            Host.Assert(0 <= iinfo && iinfo < Infos.Length);
+            return _outputType;
+        }
+
+        protected override Delegate GetGetterCore(IChannel ch, IRow input, int iinfo, out Action disposer)
+        {
+            Host.AssertValue(ch);
+            ch.AssertValue(input);
+            ch.Assert(0 <= iinfo && iinfo < Infos.Length);
+            disposer = null;
+
+            var info = Infos[iinfo];
+            if (!info.TypeSrc.IsVector)
+            {
+                throw Host.ExceptParam(nameof(input),
+                    "Text input given, expects a text vector");
+            }
+            return GetGetterVec(ch, input, iinfo);
+        }
+
+        private ValueGetter<VBuffer<float>> GetGetterVec(IChannel ch, IRow input, int iinfo)
+        {
+            Host.AssertValue(ch);
+            ch.AssertValue(input);
+            ch.Assert(0 <= iinfo && iinfo < Infos.Length);
+
+            var info = Infos[iinfo];
+            ch.Assert(info.TypeSrc.IsVector);
+            ch.Assert(info.TypeSrc.ItemType.IsText);
+
+            var srcGetter = input.GetGetter<VBuffer<DvText>>(info.Source);
+            var src = default(VBuffer<DvText>);
+            int dimension = _currentVocab.Dimension;
+            float[] wordVector = new float[_currentVocab.Dimension];
+
+            return
+                (ref VBuffer<float> dst) =>
+                {
+                    int deno = 0;
+                    srcGetter(ref src);
+                    var values = dst.Values;
+                    Utils.EnsureSize(ref values, 3 * dimension, keepOld: false);
+                    int offset = 2 * dimension;
+                    for (int i = 0; i < dimension; i++)
+                    {
+                        values[i] = int.MaxValue;
+                        values[i + dimension] = 0;
+                        values[i + offset] = int.MinValue;
+                    }
+                    for (int word = 0; word < src.Count; word++)
+                    {
+                        if (_currentVocab.GetWordVector(ref src.Values[word], wordVector))
+                        {
+                            deno++;
+                            for (int i = 0; i < dimension; i++)
+                            {
+                                float currentTerm = wordVector[i];
+                                if (values[i] > currentTerm)
```

Comment:
Can you point out the code fix for https://github.com/dotnet/machinelearning/pull/545#pullrequestreview-139596689? I'm not seeing it.

Reply:
I removed Array.Clear and replace it with setting MaxValue,0,MinValue.

---
In reply to: [204566178](https://github.com/dotnet/machinelearning/pull/545#discussion_r204566178) [](ancestors = 204566178)

Reply:
Thanks. I see it now.